ra-optimisation/search-tree-rao/src/main/java/com/farao_community/farao/search_tree_rao/SearchTreeRao.java

```diff
@@ -221,6 +221,9 @@ private RaoResult buildOutput() {
         RaoResult raoResult = new RaoResult(optimalLeaf.getStatus().equals(Leaf.Status.ERROR) ? RaoResult.Status.FAILURE : RaoResult.Status.SUCCESS);
         raoResult.setPreOptimVariantId(rootLeaf.getInitialVariantId());
         raoResult.setPostOptimVariantId(optimalLeaf.getBestVariantId());
+        if (raoResult.isSuccessful()) {
+            raoResult.setOptimizedNetwork(optimalLeaf.getRaoData().getNetwork()); //? TODO does the network in optimialLeaf contain all network actions ?
```

Comment:
No, you must add optimalLeaf.applyRangeActionResultsOnNetwork()

Reply:
OK, could you please make a correction? I will just close this PR. thanks.

Reply:
I am not sure applyRangeActionResultsOnNetwork will solve the problem.